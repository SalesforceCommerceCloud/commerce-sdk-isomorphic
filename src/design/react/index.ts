/*
 * Copyright (c) 2025, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

// Main Provider
export {
  PageDesignerProvider,
  usePageDesignerMode,
} from './context/PageDesignerProvider';

export {useDesignContext} from './context/DesignContext';
export type {DesignContextType} from './context/DesignContext';

// Components
export {createReactComponentDesignDecorator} from './components/ComponentDecorator';
export {createReactRegionDesignDecorator} from './components/RegionDecorator';
