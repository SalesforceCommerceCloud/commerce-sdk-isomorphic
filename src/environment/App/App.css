body {
  font-size: 16px;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  align-items: center;
  background-color: #282c34;
  color: white;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  justify-content: center;
  min-height: 100vh;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.search-bar {
  background: #282c34;
  color: #61dafb;
  padding: 10px 25px;
  text-align: center;
}

.search-box {
  margin: 1em;
}

.search-button {
  background: #61dafb;
  border: none;
  color: black;
}

h3 {
  color: #6d6d6d;
  font-size: 20px;
  font-style: normal;
  font-weight: 300;
}

.product-component {
  margin: 2em;
}

.product-currency {
  font-size: 12px;
  font-weight: bold;
}

.product-id {
  font-size: 12px;
}
