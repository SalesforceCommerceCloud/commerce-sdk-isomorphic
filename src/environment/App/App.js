import React from 'react';
import {Counter} from 'lib';
import classes from './App.module.css';

const App = () => (
  <div className={classes.container}>
    <Counter initialValue={0} />
  </div>
);

export default App;
