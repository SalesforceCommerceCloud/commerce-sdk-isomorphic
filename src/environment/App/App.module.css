.container {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  width: 960px;
}
