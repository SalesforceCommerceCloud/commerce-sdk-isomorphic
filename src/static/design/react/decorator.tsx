/*
 * Copyright (c) 2025, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
import React from 'react';

export function createReactDesignDecorator(): (
  component: React.ReactElement
) => React.ReactElement {
  // TODO: Implement design decorator for React components.
  return (component: React.ReactElement) => <>{component}</>;
}
