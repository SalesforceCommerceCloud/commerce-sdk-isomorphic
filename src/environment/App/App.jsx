/*
 * Copyright (c) 2021, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

/* eslint-disable react/require-default-props */
// eslint-disable-next-line no-use-before-define
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {ShopperLogin, ShopperSearch} from 'lib';
import * as slasHelper from 'lib/helpers/slasHelper';
import config from '../config';
import './App.css';

const slasClient = new ShopperLogin(config);
const searchClient = new ShopperSearch(config);

const Product = ({id = '', name = '', price = 0, currency = ''}) => (
  <div className="product-component">
    <h3>{name}</h3>
    <p className="product-price">
      ${price}
      <span className="product-currency"> {currency}</span>
    </p>
    <p className="product-id">
      ID
      {id}
    </p>
  </div>
);

Product.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  currency: PropTypes.string,
};
class App extends Component {
  constructor(props) {
    super(props);
    this.searchClient = searchClient;
    this.state = {
      searchText: 'sony',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  async componentDidMount() {
    const {searchText} = this.state;
    this.doSearch(searchText);
  }

  handleChange(event) {
    this.setState({searchText: event.target.value});
  }

  handleSubmit(event) {
    const {searchText} = this.state;
    this.doSearch(searchText);
    event.preventDefault();
  }

  async getSearchResults(text) {
    await this.getToken();
    const {token} = this.state;
    this.searchClient.clientConfig.headers.authorization = token;
    return searchClient.productSearch({
      parameters: {q: text},
    });
  }

  async getToken() {
    const authResponse = await slasHelper.loginGuestUser(slasClient, {
      redirectURI: new URL('/callback', window.location).href,
    });
    this.state.token = `Bearer ${authResponse.access_token}`;
  }

  async doSearch() {
    const {searchText} = this.state;
    const results = await this.getSearchResults(searchText);
    if (results?.hits[0]) {
      this.setState({...results.hits[0]});
    } else {
      this.setState({productId: null});
    }
  }

  render() {
    const {searchTerm, currency, productId, productName, price} = this.state;
    return (
      <div className="search-component">
        <form className="search-bar" onSubmit={this.handleSubmit}>
          <label className="search-label" htmlFor="item-search">
            Product Search
            <input
              id="item-search"
              className="search-box"
              type="text"
              value={searchTerm}
              onChange={this.handleChange}
            />
          </label>
          <input className="search-button" type="submit" value="Search" />
        </form>
        <Product
          currency={currency}
          id={productId}
          name={productName}
          price={price}
        />
      </div>
    );
  }
}

export default App;
