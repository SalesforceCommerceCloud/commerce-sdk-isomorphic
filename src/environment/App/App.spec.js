import React from 'react';
import {render} from '@testing-library/react';
import App from './App';

describe('environment > App', () => {
  it('renders without crashing', () => {
    /**
     * `asFragment`:
     * @see https://testing-library.com/docs/react-testing-library/api#asfragment
     */
    const {asFragment} = render(<App />);

    /**
     * Basic snapshot test to make sure, that rendered component
     * matches expected footprint.
     */
    expect(asFragment()).toMatchSnapshot();
  });
});
