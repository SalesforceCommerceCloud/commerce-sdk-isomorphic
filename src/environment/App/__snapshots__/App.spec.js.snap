// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`environment > App renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="counter"
    >
      <h2
        class="header"
      >
        Counter
      </h2>
      <button
        class="button"
        type="button"
      >
        Increment by one
      </button>
      <div>
        Total value: 
        <strong>
          0
        </strong>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
